src/test/java/com/mongodb/kafka/connect/source/MongoCopyDataManagerTest.java

```diff
@@ -152,34 +149,39 @@ void testAppliesTheExpectedPipelines() {
       results = asList(copyExistingDataManager.poll(), copyExistingDataManager.poll());
     }
 
-    List<Optional<BsonDocument>> expected = asList(Optional.of(result), Optional.empty());
+    List<Optional<BsonDocument>> expected = asList(createOutput(jsonTemplate), Optional.empty());
     assertEquals(expected, results);
   }
 
   @Test
   @DisplayName("test blocks adding docs to the queue")
   void testBlocksAddingResultsToTheQueue() {
-    List<RawBsonDocument> docs =
+    List<String> templates =
         IntStream.range(0, 10)
-            .mapToObj(i -> RawBsonDocument.parse(format("{'_id': {'_id': %s, 'copy': true}}", i)))
+            .mapToObj(MongoCopyDataManagerTest::createTemplate)
             .collect(Collectors.toList());
 
+    List<RawBsonDocument> inputDocs =
+        templates.stream().map(MongoCopyDataManagerTest::createInput).collect(Collectors.toList());
+
     when(mongoClient.getDatabase(TEST_DATABASE)).thenReturn(mongoDatabase);
     when(mongoDatabase.getCollection(TEST_COLLECTION, RawBsonDocument.class))
         .thenReturn(mongoCollection);
     when(mongoCollection.aggregate(anyList())).thenReturn(aggregateIterable);
     doCallRealMethod().when(aggregateIterable).forEach(any(Consumer.class));
     when(aggregateIterable.iterator()).thenReturn(cursor);
 
-    Boolean[] hasNextResponses = new Boolean[docs.size()];
+    Boolean[] hasNextResponses = new Boolean[inputDocs.size()];
     Arrays.fill(hasNextResponses, true);
     hasNextResponses[hasNextResponses.length - 1] = false;
 
     when(cursor.hasNext()).thenReturn(true, hasNextResponses);
     when(cursor.next())
         .thenReturn(
-            docs.get(0),
-            docs.subList(1, docs.size()).toArray(new RawBsonDocument[docs.size() - 1]));
+            inputDocs.get(0),
```

Comment:
I suppose it allows users to pass an array as the single item returned without it getting expanded via varargs. eg:

```
byte[] bytes = ...;
when(...)
  .thenReturn(bytes);
```

Reply:
Indeed, this is a good point! (though, for this to happen, we need `Byte[]`, not `byte[]`) However, with `thenReturn(T... value)` the same can be achieved by writing
```java
Byte[] bytes = ...;
thenReturn(new Object[] {bytes});
```
If an API user forgets (not unexpectedly) not think about varargs behavior and simply call `thenReturn(bytes)`, it would return `Byte`, not `Byte[]`. If a user does not use the returned value, the user may not notice the problem.

A situation, in which all solutions are flawed. Anyway, thanks you, now I can see why the Mockito may have the API it has.